avm/ptn/sa/modernize-your-code/main.bicep

```diff
@@ -0,0 +1,549 @@
+    managedIdentities: {
+      userAssignedResourceIds: [
+        appIdentity.outputs.resourceId
+      ]
+    }
+    appInsightsConnectionString: enableMonitoring ? applicationInsights.outputs.connectionString : null
+    appLogsConfiguration: enableMonitoring
+      ? {
+          destination: 'log-analytics'
+          logAnalyticsConfiguration: {
+            customerId: logAnalyticsWorkspace.outputs.logAnalyticsWorkspaceId
+            sharedKey: logAnalyticsWorkspace.outputs.primarySharedKey
+          }
+        }
+      : {}
+    workloadProfiles: enableRedundancy
+      ? [
+          {
+            maximumCount: 3
+            minimumCount: 3
+            name: 'CAW01'
+            workloadProfileType: 'D4'
+          }
+        ]
+      : [
+          {
+            name: 'Consumption'
+            workloadProfileType: 'Consumption'
+          }
+        ]
+    tags: allTags
+    enableTelemetry: enableTelemetry
+  }
+}
+
+module containerAppFrontend 'br/public:avm/res/app/container-app:0.17.0' = {
+  name: take('container-app-frontend-${resourcesName}-deployment', 64)
+  params: {
+    name: take('ca-${resourcesName}frontend', 32)
+    location: location
+    environmentResourceId: containerAppsEnvironment.outputs.resourceId
+    managedIdentities: {
+      userAssignedResourceIds: [
+        appIdentity.outputs.resourceId
+      ]
+    }
+    containers: [
+      {
+        env: [
+          {
+            name: 'API_URL'
+            value: 'https://${containerAppBackend.outputs.fqdn}'
+          }
+        ]
+        image: 'cmsacontainerreg.azurecr.io/cmsafrontend:latest'
+        name: 'cmsafrontend'
+        resources: {
+          cpu: '1'
+          memory: '2.0Gi'
+        }
+      }
+    ]
+    ingressTargetPort: 3000
+    ingressExternal: true
+    scaleSettings: {
+      maxReplicas: enableScaling ? 3 : 1
+      minReplicas: 1
+      rules: enableScaling
+        ? [
+            {
+              name: 'http-scaler'
+              http: {
+                metadata: {
+                  concurrentRequests: 100
+                }
+              }
+            }
+          ]
+        : []
+    }
+    tags: allTags
+    enableTelemetry: enableTelemetry
+  }
+}
+
+module containerAppBackend 'br/public:avm/res/app/container-app:0.17.0' = {
+  name: take('container-app-backend-${resourcesName}-deployment', 64)
+  #disable-next-line no-unnecessary-dependson
+  dependsOn: [applicationInsights] // required due to optional flags that could change dependency
+  params: {
+    name: take('ca-${resourcesName}backend', 32)
+    location: location
+    environmentResourceId: containerAppsEnvironment.outputs.resourceId
+    managedIdentities: {
+      userAssignedResourceIds: [
+        appIdentity.outputs.resourceId
+      ]
+    }
+    containers: [
+      {
+        name: 'cmsabackend'
+        image: 'cmsacontainerreg.azurecr.io/cmsabackend:latest'
```

Comment:
hardcoding latest image is not a good idea, specially if at some point, there is a non backward compatible change that depends on the infrastructure (e.g. depend of a new environment variable).

Recommended to create a parameter with a default value set to a specific version of the image that works for the current - e.g. `param backendContainerImageTag string = 'latest_2025-06-17_704'`